src/it/scala/deduplication/DeduplicationSuite.scala

```diff
@@ -87,6 +87,31 @@ class DeduplicationSuite extends FunSuite {
       }
   }
 
+  // TODO this test is ignored because it is the symptoms of the issue we have with this library
+  test("Deduplication should not re-process multiple event if it failed the first time".ignore) {
+
+    val processorId = given[UUID]
+    val id = given[UUID]
+
+    deduplicationResource(processorId, 1.seconds)
+      .use { ps =>
+        for {
+          ref <- Ref[IO].of(0)
+          _ <- ps
+            .protect(id, IO.raiseError[Unit](new RuntimeException("Expected exception")), IO.unit)
+            .attempt
+          _ <- List
+            .fill(100)(id)
+            .parTraverse { _ =>
+              ps.protect(id, ref.update(_ + 1), IO.unit)
+            }
+          result <- ref.get
+        } yield {
+          assertEquals(result, 1)
+        }
```

Comment:
Not sure how thi test is explicit, but it basically:
- Create a `Ref[Int]` and initialise it to 0
- Start a process and crash it but do not crash the whole thing (attempt is the reverse or rethrow, it transform an `IO[A]` to `Either[Throwable, A]`
- Start 100 processes that try to increment Ref[Int]
- Check that Ref[Int]  contains 1